### ISSUE TITLE
CrossComponentIterator.java should allow for a collection of startVertices

### ISSUE THREAD

**Issue by sharpTrick** [jgrapht/jgrapht#398]
Clearly CrossComponentIterator already makes use of a private startVertexIterator. The case where one wants to traverse an entire directed graph obviously needs multiple starting vertices in certain instances. I would argue that this is not the only use case where specifying multiple starting vertices is useful.

I specifically need to construct a sub-graph contain all vertices down wind from a specific set of vertices. To do this I need either a set of these vertices or a predicate dictating if a vertex would belong to such a set. One way to accomplish this is to collect vertices from a depth first traversal with a specified startVertexIterator relating to the specific set of vertices.

I believe that this should be a very simple fix that will not break anything. I think its as simple as adding an additional constructor to CrossComponentIterator and its sub classes to take an iterable or iterator.

**Comment by d-michail** [comment]
Adding a constructor of the form 
```
public CrossComponentIterator(Graph<V, E> g, Iterable<V> startVertices)
```
does indeed makes sense. 

It should however respect the semantics of the iterator and not break current functionality. Note that it will need additional checks to make sure that the vertices returned by the iterator belong to the graph, etc.

Would you care to give it a try and submit a PR?

**Comment by sharpTrick** [comment]
Indeed

**Comment by sharpTrick** [comment]
Since I added a a constructor of the sort Constructor(graph, iterable<V>), I introduced an error that manifests with every use of Constructor(graph, null) as Constructor(graph, iterable<V>) and already present Constructor(graph, v) both match the uncast null.

Evidently I missed one of these instances in the code. Unfortunately this error will propagate into any one's codebase that used this construction pattern.

In any event I have made sure that Constructor(graph, (V) null) and Constructor(graph, (iterable<V>) null) result in the same outcome as Constructor(graph). 

It is my opinion that it would be better to do away with the ability to use null as a constructor argument since it has the same meaning as not abstaining from the use of said argument.

I will take care of the errors I missed. 

@d-michail, please advise:
options:
    - continue to allow for null argument (users will have fix any use of such a constructor)
    - make constructors require non-null arguments (users will have to fix any use of such a constructor)
    - change the constructor to remove the ambiguity between the two (poor style but users unscathed)
        - reverse the order of graph and iterable
        - add some extra argument

**Comment by jsichi** [comment]
The "poor style" options are both unappealing, so I don't see a clean way to go our usual deprecation route here.

If I've understood #401 correctly, apps that were passing null explicitly will still be binary compatible after this change; they'll get a compiler ambiguity error when attempting to build.  I think this should be such a rare enough possibility that we can live with it.

**Comment by sharpTrick** [comment]
Awesome, then I believe the PR will work as is.

**Comment by d-michail** [comment]
Resolved in #401.
